browser/ui/webui/brave_welcome_ui.cc

```diff
@@ -48,20 +52,77 @@ void OpenJapanWelcomePage(Profile* profile) {
   }
 }
 
+// Returns whether the P3A opt-in prompt should be shown.
+bool IsP3AOptInEnabled() {
+  bool enabled = false;
+  // Check the local_state pref for study status.
+  PrefService* local_state = g_browser_process->local_state();
+  if (local_state->FindPreference(brave::kP3AOptInEnabled)->IsDefaultValue()) {
+    // Decide on first run if an install should participate in opt_in.
+    if (base::RandDouble() < 0.05) {
```

Comment:
Why we doing it on the client side, not using Griffin?

Reply:
We need to determine study participation at first-run time. As I understand it, the Griffin seed isn't downloaded and active until at least the second run. #15229 proposes shipping a better initial seed with the installer, but the study was high priority, so I used a client-side hack.

We plan to remove all of this after a couple of releases.

Reply:
I added a comment to clarify the motivation.